### PR TITLE
Dl image zoom modal

### LINKED ISSUE #1240: [website addon] ImageComponent
- [x] add modal functionality, see IDL site for example https://www.ibm.com/design/language/elements/color-ui
- [x] Move  from `/src/components/` to `/packages/addons-website/src/`
- [x] Update add-on storybook
- [x] Update imports inside website to pull from add on instead of components folder.

### PR DESCRIPTION
**Pull request by mizj**
#### Changelog

**New**

- In "addons-website" added Image Zoom Modal for DesignLanguage site
- added Overlay component which is used by the Image Zoom Modal

closes #1240

### CODE REVIEW

File: packages/addons-website/src/components/ImageZoomModal/ImageZoomModal.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import classnames from 'classnames';
+import { settings } from 'carbon-components';
+
+import Overlay from '../Overlay'
+
+import { Icon } from 'carbon-components-react';
+
+const { prefix } = settings;
+
+const MagnifierIcon = (
```
[the remark this answers is not in the document]
Comment (mizj) [reply packages/addons-website/src/components/ImageZoomModal/ImageZoomModal.js]: OK will do.

[the remark this answers is not in the document]
Comment (mizj) [reply packages/addons-website/src/components/ImageZoomModal/ImageZoomModal.js]: @alisonjoseph - Do you want me to keep the copy in 'addons-website' or do you want to move it entirely?

[the remark this answers is not in the document]
Comment (alisonjoseph) [reply packages/addons-website/src/components/ImageZoomModal/ImageZoomModal.js]: I'm not sure exactly what you mean?

[the remark this answers is not in the document]
Comment (mizj) [reply packages/addons-website/src/components/ImageZoomModal/ImageZoomModal.js]: Ok I will move the ImageComponent to addons and add the additional functionality to it to zoom.

**Comment by alisonjoseph** [comment]
Looking good! 

Can you update the import on line `30` of `gatsby-config.js` to be `import { ImageComponent } from '@carbon/addons-website';` so its pulling from the addons

You'll also need to add an export to `carbon-website/packages/addons-website/src/index.js`
```
export { default as ImageComponent } from './components/ImageComponent';
```

and also add to the following to `carbon-website/packages/addons-website/src/components/ImageComponent/index.js`
```
import ImageComponent from './ImageComponent';
export default ImageComponent;
```

then double check that the build works, `cd packages/addons-website` `yarn build`

**Comment by alisonjoseph** [comment]
Can you fix z-index? I added to a test page in Carbon and it sits below the shell/nav

Screen Shot 2019-03-13 at 1 14 45 PM

(would also be nice to add a gray-10 background to images, since we use many transparent images)

**Comment by alisonjoseph** [comment]
Can we have the `caption` show up as the title if it doesn't already? and add to example in storybook

File: packages/addons-website/scss/components/_overlay.scss [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+.#{$prefix}--overlay-image {
+  height: auto;
+  display:block;
+  margin:auto;
+  max-height: 100%;
+  max-width: 100%;
+  object-fit: contain;
+  user-select: none;
+}
```
Comment (alisonjoseph) [inline packages/addons-website/scss/components/_overlay.scss]: Can we add `background: $ui-background` here, then should be good to go! 🎉

File: src/content/components/accordion/style.mdx [hunk 3]
```diff
@@ -37,7 +37,7 @@ There is no limit to the height of an open row, however, follow the other paddin
 | `.bx--accordion__content`      | padding-left               | 16 / 1     | $spacing-05 |
 | `.bx--accordion__item--active` | padding-bottom             | 24 / 1.5   | $spacing-06 |
 
-<ImageComponent caption="Structure and spacing measurements for accordion | px / rem" fixed="default">
+<ImageComponent caption="Structure and spacing measurements for accordion | px / rem" fixed="default" zoom="true">
```
Comment (alisonjoseph) [inline src/content/components/accordion/style.mdx]: we should revert this now that we know its working

File: packages/addons-website/scss/components/_overlay.scss (hunk 2 above)
Comment (mizj) [reply packages/addons-website/scss/components/_overlay.scss]: updated.

File: src/content/components/accordion/style.mdx (hunk 3 above)
Comment (mizj) [reply src/content/components/accordion/style.mdx]: Done.

**Pull request merged**